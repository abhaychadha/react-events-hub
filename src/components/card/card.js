import React from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography'
import Facebook from '../facebook';

const CardBlock = props => {
    const {classes} = props;
    
    return <Card className={classes.card} md={12}>      
        <CardMedia
          component="img"
          alt="No Image"
          className={classes.media}
          height="140"
          image={props.imageUrl}
          title=""
        />
        <CardContent className={classes.cardContent}>
          <Typography gutterBottom variant="h4" component="h2">
            {props.title}
            <Facebook social={props.social}/>
          </Typography>
          
          <Typography gutterBottom variant="h5" component="h2">
            {props.body}
          </Typography>
          <a size="small" color="primary" href={props.link} rel="noopener noreferrer" target="_blank" className={classes.linkUrl}>
            Learn More
          </a>
        </CardContent>
    </Card>
}

export default CardBlock;