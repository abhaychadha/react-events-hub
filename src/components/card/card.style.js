export const styles = {
  card: {
    maxWidth: 345,
  },
  media: {  
    objectFit: 'cover',
  },
  linkUrl: {
    display: "block",
    textAlign: "left",
    fontSize: "10px",
    cursor: "pointer"
  }
};