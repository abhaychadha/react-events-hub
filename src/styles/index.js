import { injectGlobal } from "styled-components";
import common from "./common";
import fonts from "./fonts.css";
injectGlobal`  
  ${fonts}
  ${common}
`;
