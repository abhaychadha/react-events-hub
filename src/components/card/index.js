import Card from "./card";
import { withStyles } from '@material-ui/core/styles';
import {styles} from './card.style';

export default withStyles(styles)(Card);