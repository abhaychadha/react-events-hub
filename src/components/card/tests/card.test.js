import React from 'react';
import Card from "../card";
import { withStyles } from '@material-ui/core/styles';
import {styles} from '../card.style';
import { createShallow } from '@material-ui/core/test-utils';

const StyledCard = withStyles(styles)(Card);

describe('StyledCard', () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow(<StyledCard />);
  });

  it('Card renders with style', () => {
    const wrapper = shallow(<StyledCard />);
    expect(wrapper).toMatchSnapshot();
  });
});