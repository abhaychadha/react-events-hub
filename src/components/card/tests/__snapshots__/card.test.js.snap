// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledCard Card renders with style 1`] = `
<CardBlock
  classes={
    Object {
      "card": "CardBlock-card-1",
      "linkUrl": "CardBlock-linkUrl-3",
      "media": "CardBlock-media-2",
    }
  }
/>
`;
